src/NzbDrone.Mono/ProcMountProvider.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Extensions;
+using Mono.Unix;
+
+namespace NzbDrone.Mono
+{
+    public interface IProcMountProvider
+    {
+        List<IMount> GetMounts();
+    }
+
+    public class ProcMountProvider : IProcMountProvider
+    {
+        private static string[] _fixedTypes = "ext3,ext2,ext4,vfat,fuseblk,xfs,jfs,msdos,ntfs,minix,hfs,hfsplus,qnx4,ufs,btrfs".Split(',');
```

Comment:
Torn on this, bit shorter than building it manually and less work, but it feels a bit odd still.